contentcuration/contentcuration/frontend/channelEdit/vuex/clipboard/actions.js

```diff
@@ -89,76 +90,150 @@ export function loadClipboardNodes(context, { parent, ancestorId }) {
-            if (!(node.id in context.state.selected)) {
-              context.commit('UPDATE_SELECTION_STATE', {
-                id: node.id,
-                selectionState: SelectionFlags.NONE,
-                ancestorId,
-              });
-            }
-          });
-          return [];
+  }
+
+  // Since this must be a new clipboard node that's not the top-level ancestor, we'll need
+  // to find the real parent (which should already be loaded into contentNode Vuex state) and use
+  // it to load its real children. We also get the ancestor node, for uniquely referencing this
+  // copy of the source node in the clipboard
+  const contentNode = context.getters.getContentNodeForRender(parent);
+  const ancestor = context.getters.getClipboardAncestorNode(parent);
+  if (!ancestor) {
+    throw new Error('Unexpected null ancestor');
+  }
+
+  if (contentNode && contentNode.resource_count > 0) {
+    return context
+      .dispatch('contentNode/loadContentNodes', { parent: contentNode.id }, { root: true })
+      .then(nodes => {
+        // Ensure we don't add excluded nodes to our Vuex map
+        nodes = nodes
+          .filter(node => !isExcludedNode(ancestor, node))
+          .map(node => ({
+            // For new version clipboard nodes, we don't need `node.id` for anything
+            // so we can change it, and the clipboard should not be manipulating the
+            // source node anyway, normally requiring its ID
+            id: selectionId(node.id, ancestor.id),
+            kind: node.kind,
+            parent,
+            lft: node.lft,
+            source_node_id: node.node_id,
+            source_channel_id: node.channel_id,
+            total_count: node.total_count,
+            resource_count: node.resource_count,
+            extra_fields: {
+              [ClipboardNodeFlag]: true,
+            },
+          }));
+
+        return context.dispatch('addClipboardNodes', {
+          nodes,
+          parent,
         });
-    }
+      });
   }
   return [];
 }
 
+export function addClipboardNodes(context, { nodes, parent }) {
+  const isRootInsert = parent === context.rootGetters['clipboardRootId'];
+  nodes.forEach(node => {
+    // Add selection state
+    if (!(node.id in context.state.selected)) {
+      // This is referencing what the "parent" is in the clipboard UI for determining
+      // selection inheritance
+      const selectionParent = isRootInsert ? node.source_channel_id : node.parent;
+
+      // Determine parent selection state so we can add new descendant nodes
+      // with the appropriate selection state
+      const parentSelection = context.getters.currentSelectionState(selectionParent);
+
+      context.commit('UPDATE_SELECTION_STATE', {
+        id: node.id,
+        selectionState:
+          parentSelection & SelectionFlags.ALL_DESCENDANTS ? SELECT_ALL : SelectionFlags.NONE,
+      });
+    }
+  });
+
+  context.commit('ADD_CLIPBOARD_NODES', nodes);
+  return nodes;
+}
```

Comment:
Dedicated action to ensure that the selection state of a node is initialized correctly, according to its parent selection state.